ui/src/pages/signUpPage.ts

```diff
@@ -0,0 +1,56 @@
+import { SIGN_UP_URL, confirmationMessage, defaultTimer, lengthErrorElement, takenUsernameErrorElement } from "../support/constants";
+import { SIGN_UP_FIELDS } from "../support/types";
+import { BasePage } from "./basePage";
+import { Page } from "@playwright/test";
+
+export class SignUpPage extends BasePage {
+  constructor(page: Page) {
+    super(page)
+
+    this.url = SIGN_UP_URL
+  }
+
+
+  public async getInputFieldByName(field: SIGN_UP_FIELDS) {
+    return this.page.locator(`#new_user_${field}`)
+  }
+
+
+  public getContinueButton() {
+    return this.page.locator(`button[data-qa-selector = "new_user_register_button"]`)
+  }
+
+  public getLengthError() {
+    return this.page.locator(lengthErrorElement)
+  }
+
+  public getUsernameError() {
+    return this.page.locator(takenUsernameErrorElement)
+  }
+
+  public getConfirmationMessage() {
+    return this.page.locator(confirmationMessage)
+  }
+
+  public async fillFirstNameField(firstName: string,) {
+    await (await this.getInputFieldByName(SIGN_UP_FIELDS.FIRST_NAME)).type(firstName, { delay: defaultTimer });
+  }
```

Comment:
I don't see any great necessity to create a wrapper method over `await (await this.getInputFieldByName(SIGN_UP_FIELDS.FIRST_NAME)).type(firstName, { delay: defaultTimer });` action. Please delete it.

Reply:
done